test_iokit.py

```diff
@@ -122,3 +159,206 @@ def test_IOObjectCopySuperclassForClass():
+        "IOService".encode(),
+        None
+    )
+    assert err == 0
+    assert location.replace(b"\x00", b"") == b"0"
+
+
+def test_IORegistryEntryGetPath():
+    device = _get_IOResources()
+
+    err, path = iokit.IORegistryEntryGetPath (
+        device,
+        "IOService".encode(),
+        None
+    )
+    assert err == 0
+    assert path.replace(b"\x00", b"") == b"IOService:/IOResources"
+
+
+def test_IORegistryEntryCopyPath():
+    device = _get_IOResources()
+
+    path = iokit.IORegistryEntryCopyPath(device, "IOService".encode())
+    assert path == "IOService:/IOResources"
+
+
+def test_IORegistryEntryIDMatching():
+    device = iokit.IORegistryEntryIDMatching(4294967578)
+    assert iokit.corefoundation_to_native (device) == {
+        "IORegistryEntryID": 4294967578
+    }
+
+
+def test_IORegistryEntryGetRegistryEntryID():
+    device = _get_IOResources()
+    
+    err, obj_id = iokit.IORegistryEntryGetRegistryEntryID (
+            device,
+            None
+        )
+    assert err == 0
+    assert obj_id == 4294967578
+
+
+def test_IORegistryEntryGetChildIterator():
+    device = _get_IOResources()
+    
+    err, iterator = iokit.IORegistryEntryGetChildIterator (
+            device,
+            "IOService".encode(),
+            None
+        )
+    assert err == 0
+
+    interface = list(iokit.ioiterator_to_list(iterator))
+
+    err, props = iokit.IORegistryEntryCreateCFProperties (
+            interface[0], None, iokit.kCFAllocatorDefault, iokit.kNilOptions
+        )
+    assert err == 0
+    assert props.get("IOProviderClass") == "IOResources"
+
+    for i in interface:
+        err = iokit.IOObjectRelease(i)
+        assert err == 0
+
+
+def test_IORegistryCreateIterator():
+    err, iterator = iokit.IORegistryCreateIterator (
+            iokit.kIOMainPortDefault,
+            "IODeviceTree".encode(),
+            iokit.kIORegistryIterateRecursively,
+            None
+        )
+    assert err == 0
+
+    interface = list(iokit.ioiterator_to_list(iterator))
+
+    err, props = iokit.IORegistryEntryCreateCFProperties (
+            interface[0], None, iokit.kCFAllocatorDefault, iokit.kNilOptions
+        )
+    assert err == 0
+    assert props.get("IOPlatformUUID") == "115C2C75-F539-52BE-90AF-EBBC65ED6CB8"
+
+    for i in interface:
+        err = iokit.IOObjectRelease(i)
+        assert err == 0
+
+
+def test_IORegistryEntryCreateIterator():
+    device = _get_IOResources()
+
+    err, iterator = iokit.IORegistryEntryCreateIterator (
+            device,
+            "IOService".encode(),
+            iokit.kIORegistryIterateRecursively,
+            None
+        )
+    assert err == 0
+
+    interface = list(iokit.ioiterator_to_list(iterator))
+
+    err, props = iokit.IORegistryEntryCreateCFProperties (
+            interface[0], None, iokit.kCFAllocatorDefault, iokit.kNilOptions
+        )
+    assert err == 0
+    assert props.get("IOProviderClass") == "IOResources"
+
+    for i in interface:
```

Comment:
Release device and iterator

Reply:
See [here](https://github.com/USBToolBox/tool/pull/60#discussion_r1248039736)